client/src/components/CinemaHall/index.js

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-plusplus */
+import React, { useContext, useEffect } from 'react';
+import Seat from '../Seat';
+import './style.scss';
+import { ReservationContext } from '../../context/Reservation';
+import useFetchedData from '../../hooks/useFetchedData';
+
+const CinemaHall = () => {
+  const [reservation, dispatch] = useContext(ReservationContext);
+  const { cinemaHall } = reservation;
+  const mongoCinemaHall = useFetchedData('api/cinemahalls');
+
+  useEffect(() => {
+    dispatch({
```

Comment:
It should only be dispatched in case the action is resolved. Perhaps you can outsource it to cinemaHall actions file?